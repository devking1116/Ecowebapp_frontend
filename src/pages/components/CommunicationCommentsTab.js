/* @flow */
import React from 'react';
import { Feed, Icon, Label } from 'semantic-ui-react';
import reactCSS from 'reactcss';

import { LogbookLabel, LogbookLabelGroup } from '../../logbook';
import { Comment } from '../../core';

type Props = {
  //TODO, changes this to a list of comment ids, and connect each comment to the redux state
  comments?: Array<{
    author: string,
    date: string,
    body: string,
    status?: 'active' | 'in-active',
    actionsTakens: Array<string>,
    linkedLogbooks?: Array<string>,
    attachments?: Array<{
      //TODO, get info for attachment previews
    }>,
  }>,
};

/**
 * The component that displays the comments tab on the Communication Page,
 */
export default function CommunicationCommentsTab({ comments, ...otherProps }: Props): React.Element<*> {

  const styles = reactCSS({
    'default': {
      commentsWrapper: {
        paddingTop: 20,
      },
    },
  });

  return (
    <Feed {...otherProps} style={styles.commentsWrapper}>
    {comments? comments.map(({ linkedLogbooks, attachments, ...comment }, index) =>
      <Comment {...comment} key={index}>
      {attachments?
        <span>
        {attachments.length === 1?
          'Attachment':
          `${attachments.length} Attachments`
        }
        {attachments.map(({}, index) =>
          <Icon name="file" key={index} />
        )}
        </span>:
        null
      }
      {linkedLogbooks?
        <LogbookLabelGroup>
        {linkedLogbooks.map((linkedLogbook, index) =>
          <LogbookLabel key={index}>{linkedLogbook}</LogbookLabel>
        )}
        </LogbookLabelGroup>:
        null
      }
      </Comment>
    ): null}
    </Feed>
  );
}
