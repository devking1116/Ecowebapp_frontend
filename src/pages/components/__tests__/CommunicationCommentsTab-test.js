import React from 'react';
import renderer from 'react-test-renderer';

import CommunicationCommentsTab from '../CommunicationCommentsTab';

const testComments = [
  {
    author: 'Richard Simms',
    date: 'Today 5:45 am',
    body: 'Test comment 0',
    status: 'active',
    actionsTakens: ['Spoke to Resident', 'Jetted Sweer Line'],
    linkedLogbooks: [ 'logbook A' ],
  },
  {
    author: 'User 1',
    date: '1/1/2016',
    body: 'Test comment 1',
    linkedLogbook: [ 'logbook 0', 'logbook 1' ],
    attachments: [
      {},
    ],
  },
  {
    author: 'User 2',
    date: '2/2/2016',
    body: 'Test comment 2',
  },
  {
    author: 'User 3',
    date: '3/3/2016',
    body: 'Test comment 3',
    status: 'in-active',
    attachments: [
      {},
      {},
      {},
    ],
  },
];

test('it displays the given comments.', () => {
  const component = renderer.create(
    <CommunicationCommentsTab comments={testComments}/>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
