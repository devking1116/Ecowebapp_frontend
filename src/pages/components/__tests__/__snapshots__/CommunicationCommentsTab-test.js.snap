exports[`test it displays the given comments. 1`] = `
<div
  className="ui feed"
  style={
    Object {
      "paddingTop": 20,
    }
  }>
  <div
    className="event">
    <div
      className="label"
      style={
        Object {
          "position": "relative",
        }
      }>
      <img
        src="http://semantic-ui.com/images/avatar/small/elliot.jpg" />
      <i
        aria-hidden="true"
        className="green circle icon"
        style={
          Object {
            "fontSize": 10,
            "marginLeft": -20,
            "marginTop": 20,
            "position": "absolute",
          }
        } />
    </div>
    <div
      className="content"
      style={
        Object {
          "marginTop": -15,
        }
      }>
      <div
        className="meta"
        style={
          Object {
            "width": "100%",
          }
        }>
        <div
          style={
            Object {
              "borderRight": "1px solid #d4d4d5",
              "display": "table-cell",
              "fontSize": 14,
              "width": 200,
            }
          }>
          <span
            style={
              Object {
                "color": "#4183c4",
              }
            }>
            Richard Simms
          </span>
          <br />
          <span>
            Today 5:45 am
          </span>
        </div>
        <div
          style={
            Object {
              "display": "table-cell",
              "width": 100,
            }
          }>
          <i
            aria-hidden="true"
            className="green power icon"
            style={
              Object {
                "fontSize": 14,
                "paddingLeft": 18,
                "paddingTop": 10,
                "position": "absolute",
                "zIndex": "1001",
              }
            } />
          <div
            aria-busy={undefined}
            aria-disabled={undefined}
            aria-expanded={false}
            aria-multiselectable={undefined}
            className="ui selection dropdown"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            role="listbox"
            style={
              Object {
                "border": "none",
                "fontSize": 14,
                "paddingLeft": 42,
              }
            }
            tabIndex={0}>
            <select
              aria-hidden="true"
              multiple={undefined}
              name={undefined}
              type="hidden"
              value="active">
              <option
                value="" />
              <option
                value="active">
                Active
              </option>
              <option
                value="in-active">
                In-Active
              </option>
              <option
                value="none">
                None
              </option>
            </select>
            <div
              className="text">
              Active
            </div>
            <div
              className="menu transition">
              <div
                aria-checked={true}
                aria-disabled={undefined}
                aria-selected={true}
                className="active selected item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }>
                <i
                  aria-hidden="true"
                  className="green power icon" />
                Active
              </div>
              <div
                aria-checked={false}
                aria-disabled={undefined}
                aria-selected={false}
                className="item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }>
                <i
                  aria-hidden="true"
                  className="red power icon" />
                In-Active
              </div>
              <div
                aria-checked={false}
                aria-disabled={undefined}
                aria-selected={false}
                className="item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }>
                <i
                  aria-hidden="true"
                  className="grey power icon" />
                None
              </div>
            </div>
          </div>
        </div>
        <div
          aria-busy={undefined}
          aria-disabled={undefined}
          aria-expanded={false}
          aria-multiselectable={undefined}
          className="ui pointing link item top right dropdown"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          style={
            Object {
              "float": "right",
              "fontSize": 20,
              "marginTop": -30,
              "paddingRight": 15,
            }
          }
          tabIndex={0}>
          <div
            className="text" />
          <i
            aria-hidden="true"
            className="angle down icon" />
          <div
            className="menu transition">
            <div
              aria-checked={undefined}
              aria-disabled={undefined}
              aria-selected={undefined}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "color": "#4183c4",
                }
              }>
              Edit
            </div>
            <div
              aria-checked={undefined}
              aria-disabled={undefined}
              aria-selected={undefined}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "color": "#4183c4",
                }
              }>
              Link
            </div>
            <div
              aria-checked={undefined}
              aria-disabled={undefined}
              aria-selected={undefined}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "color": "#4183c4",
                }
              }>
              Share
            </div>
            <div
              aria-checked={undefined}
              aria-disabled={undefined}
              aria-selected={undefined}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "color": "#4183c4",
                }
              }>
              Add to Logbooks
            </div>
            <div
              className="divider" />
            <div
              aria-checked={undefined}
              aria-disabled={undefined}
              aria-selected={undefined}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "color": "red",
                }
              }>
              Delete
            </div>
          </div>
        </div>
      </div>
      <div
        className="meta"
        style={
          Object {
            "display": "block",
            "fontSize": 14,
          }
        }>
        <span>
          Actions Taken
        </span>
        <span
          style={
            Object {
              "border": "1px solid #d4d4d5",
              "fontSize": 14,
              "marginLeft": 10,
              "paddingLeft": 2,
              "paddingRight": 2,
            }
          }>
          Spoke to Resident
        </span>
        <span
          style={
            Object {
              "border": "1px solid #d4d4d5",
              "fontSize": 14,
              "marginLeft": 10,
              "paddingLeft": 2,
              "paddingRight": 2,
            }
          }>
          Jetted Sweer Line
        </span>
      </div>
      <div
        className="text extra">
        Test comment 0
      </div>
      <div
        className="meta"
        style={
          Object {
            "marginTop": 5,
          }
        }>
        <span
          onMouseOut={[Function]}
          onMouseOver={[Function]}>
          <div
            style={
              Object {
                "display": "flex",
                "marginTop": 5,
              }
            }>
            <div
              style={
                Object {
                  "marginLeft": 5,
                  "position": "relative",
                  "zIndex": 1,
                }
              }>
              <div
                className="ui green tiny basic label"
                onClick={[Function]}>
                logbook A
              </div>
            </div>
          </div>
        </span>
      </div>
    </div>
  </div>
  <div
    className="event">
    <div
      className="label"
      style={
        Object {
          "position": "relative",
        }
      }>
      <img
        src="http://semantic-ui.com/images/avatar/small/elliot.jpg" />
    </div>
    <div
      className="content"
      style={
        Object {
          "marginTop": -15,
        }
      }>
      <div
        className="meta"
        style={
          Object {
            "width": "100%",
          }
        }>
        <div
          style={
            Object {
              "borderRight": "1px solid #d4d4d5",
              "display": "table-cell",
              "fontSize": 14,
              "width": 200,
            }
          }>
          <span
            style={
              Object {
                "color": "#4183c4",
              }
            }>
            User 1
          </span>
          <br />
          <span>
            1/1/2016
          </span>
        </div>
        <div
          style={
            Object {
              "display": "table-cell",
              "width": 100,
            }
          }>
          <i
            aria-hidden="true"
            className="grey power icon"
            style={
              Object {
                "fontSize": 14,
                "paddingLeft": 18,
                "paddingTop": 10,
                "position": "absolute",
                "zIndex": "1001",
              }
            } />
          <div
            aria-busy={undefined}
            aria-disabled={undefined}
            aria-expanded={false}
            aria-multiselectable={undefined}
            className="ui selection dropdown"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            role="listbox"
            style={
              Object {
                "border": "none",
                "fontSize": 14,
                "paddingLeft": 42,
              }
            }
            tabIndex={0}>
            <select
              aria-hidden="true"
              multiple={undefined}
              name={undefined}
              type="hidden"
              value="none">
              <option
                value="" />
              <option
                value="active">
                Active
              </option>
              <option
                value="in-active">
                In-Active
              </option>
              <option
                value="none">
                None
              </option>
            </select>
            <div
              className="text">
              None
            </div>
            <div
              className="menu transition">
              <div
                aria-checked={false}
                aria-disabled={undefined}
                aria-selected={false}
                className="item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }>
                <i
                  aria-hidden="true"
                  className="green power icon" />
                Active
              </div>
              <div
                aria-checked={false}
                aria-disabled={undefined}
                aria-selected={false}
                className="item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }>
                <i
                  aria-hidden="true"
                  className="red power icon" />
                In-Active
              </div>
              <div
                aria-checked={true}
                aria-disabled={undefined}
                aria-selected={true}
                className="active selected item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }>
                <i
                  aria-hidden="true"
                  className="grey power icon" />
                None
              </div>
            </div>
          </div>
        </div>
        <div
          aria-busy={undefined}
          aria-disabled={undefined}
          aria-expanded={false}
          aria-multiselectable={undefined}
          className="ui pointing link item top right dropdown"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          style={
            Object {
              "float": "right",
              "fontSize": 20,
              "marginTop": -30,
              "paddingRight": 15,
            }
          }
          tabIndex={0}>
          <div
            className="text" />
          <i
            aria-hidden="true"
            className="angle down icon" />
          <div
            className="menu transition">
            <div
              aria-checked={undefined}
              aria-disabled={undefined}
              aria-selected={undefined}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "color": "#4183c4",
                }
              }>
              Edit
            </div>
            <div
              aria-checked={undefined}
              aria-disabled={undefined}
              aria-selected={undefined}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "color": "#4183c4",
                }
              }>
              Link
            </div>
            <div
              aria-checked={undefined}
              aria-disabled={undefined}
              aria-selected={undefined}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "color": "#4183c4",
                }
              }>
              Share
            </div>
            <div
              aria-checked={undefined}
              aria-disabled={undefined}
              aria-selected={undefined}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "color": "#4183c4",
                }
              }>
              Add to Logbooks
            </div>
            <div
              className="divider" />
            <div
              aria-checked={undefined}
              aria-disabled={undefined}
              aria-selected={undefined}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "color": "red",
                }
              }>
              Delete
            </div>
          </div>
        </div>
      </div>
      <div
        className="meta"
        style={
          Object {
            "display": "block",
            "fontSize": 14,
          }
        }>
        <span>
          Actions Taken
        </span>
      </div>
      <div
        className="text extra">
        Test comment 1
      </div>
      <div
        className="meta"
        style={
          Object {
            "marginTop": 5,
          }
        }>
        <span>
          Attachment
          <i
            aria-hidden="true"
            className="file icon" />
        </span>
      </div>
    </div>
  </div>
  <div
    className="event">
    <div
      className="label"
      style={
        Object {
          "position": "relative",
        }
      }>
      <img
        src="http://semantic-ui.com/images/avatar/small/elliot.jpg" />
    </div>
    <div
      className="content"
      style={
        Object {
          "marginTop": -15,
        }
      }>
      <div
        className="meta"
        style={
          Object {
            "width": "100%",
          }
        }>
        <div
          style={
            Object {
              "borderRight": "1px solid #d4d4d5",
              "display": "table-cell",
              "fontSize": 14,
              "width": 200,
            }
          }>
          <span
            style={
              Object {
                "color": "#4183c4",
              }
            }>
            User 2
          </span>
          <br />
          <span>
            2/2/2016
          </span>
        </div>
        <div
          style={
            Object {
              "display": "table-cell",
              "width": 100,
            }
          }>
          <i
            aria-hidden="true"
            className="grey power icon"
            style={
              Object {
                "fontSize": 14,
                "paddingLeft": 18,
                "paddingTop": 10,
                "position": "absolute",
                "zIndex": "1001",
              }
            } />
          <div
            aria-busy={undefined}
            aria-disabled={undefined}
            aria-expanded={false}
            aria-multiselectable={undefined}
            className="ui selection dropdown"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            role="listbox"
            style={
              Object {
                "border": "none",
                "fontSize": 14,
                "paddingLeft": 42,
              }
            }
            tabIndex={0}>
            <select
              aria-hidden="true"
              multiple={undefined}
              name={undefined}
              type="hidden"
              value="none">
              <option
                value="" />
              <option
                value="active">
                Active
              </option>
              <option
                value="in-active">
                In-Active
              </option>
              <option
                value="none">
                None
              </option>
            </select>
            <div
              className="text">
              None
            </div>
            <div
              className="menu transition">
              <div
                aria-checked={false}
                aria-disabled={undefined}
                aria-selected={false}
                className="item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }>
                <i
                  aria-hidden="true"
                  className="green power icon" />
                Active
              </div>
              <div
                aria-checked={false}
                aria-disabled={undefined}
                aria-selected={false}
                className="item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }>
                <i
                  aria-hidden="true"
                  className="red power icon" />
                In-Active
              </div>
              <div
                aria-checked={true}
                aria-disabled={undefined}
                aria-selected={true}
                className="active selected item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }>
                <i
                  aria-hidden="true"
                  className="grey power icon" />
                None
              </div>
            </div>
          </div>
        </div>
        <div
          aria-busy={undefined}
          aria-disabled={undefined}
          aria-expanded={false}
          aria-multiselectable={undefined}
          className="ui pointing link item top right dropdown"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          style={
            Object {
              "float": "right",
              "fontSize": 20,
              "marginTop": -30,
              "paddingRight": 15,
            }
          }
          tabIndex={0}>
          <div
            className="text" />
          <i
            aria-hidden="true"
            className="angle down icon" />
          <div
            className="menu transition">
            <div
              aria-checked={undefined}
              aria-disabled={undefined}
              aria-selected={undefined}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "color": "#4183c4",
                }
              }>
              Edit
            </div>
            <div
              aria-checked={undefined}
              aria-disabled={undefined}
              aria-selected={undefined}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "color": "#4183c4",
                }
              }>
              Link
            </div>
            <div
              aria-checked={undefined}
              aria-disabled={undefined}
              aria-selected={undefined}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "color": "#4183c4",
                }
              }>
              Share
            </div>
            <div
              aria-checked={undefined}
              aria-disabled={undefined}
              aria-selected={undefined}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "color": "#4183c4",
                }
              }>
              Add to Logbooks
            </div>
            <div
              className="divider" />
            <div
              aria-checked={undefined}
              aria-disabled={undefined}
              aria-selected={undefined}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "color": "red",
                }
              }>
              Delete
            </div>
          </div>
        </div>
      </div>
      <div
        className="meta"
        style={
          Object {
            "display": "block",
            "fontSize": 14,
          }
        }>
        <span>
          Actions Taken
        </span>
      </div>
      <div
        className="text extra">
        Test comment 2
      </div>
      <div
        className="meta"
        style={
          Object {
            "marginTop": 5,
          }
        } />
    </div>
  </div>
  <div
    className="event">
    <div
      className="label"
      style={
        Object {
          "position": "relative",
        }
      }>
      <img
        src="http://semantic-ui.com/images/avatar/small/elliot.jpg" />
      <i
        aria-hidden="true"
        className="red circle icon"
        style={
          Object {
            "fontSize": 10,
            "marginLeft": -20,
            "marginTop": 20,
            "position": "absolute",
          }
        } />
    </div>
    <div
      className="content"
      style={
        Object {
          "marginTop": -15,
        }
      }>
      <div
        className="meta"
        style={
          Object {
            "width": "100%",
          }
        }>
        <div
          style={
            Object {
              "borderRight": "1px solid #d4d4d5",
              "display": "table-cell",
              "fontSize": 14,
              "width": 200,
            }
          }>
          <span
            style={
              Object {
                "color": "#4183c4",
              }
            }>
            User 3
          </span>
          <br />
          <span>
            3/3/2016
          </span>
        </div>
        <div
          style={
            Object {
              "display": "table-cell",
              "width": 100,
            }
          }>
          <i
            aria-hidden="true"
            className="red power icon"
            style={
              Object {
                "fontSize": 14,
                "paddingLeft": 18,
                "paddingTop": 10,
                "position": "absolute",
                "zIndex": "1001",
              }
            } />
          <div
            aria-busy={undefined}
            aria-disabled={undefined}
            aria-expanded={false}
            aria-multiselectable={undefined}
            className="ui selection dropdown"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            role="listbox"
            style={
              Object {
                "border": "none",
                "fontSize": 14,
                "paddingLeft": 42,
              }
            }
            tabIndex={0}>
            <select
              aria-hidden="true"
              multiple={undefined}
              name={undefined}
              type="hidden"
              value="in-active">
              <option
                value="" />
              <option
                value="active">
                Active
              </option>
              <option
                value="in-active">
                In-Active
              </option>
              <option
                value="none">
                None
              </option>
            </select>
            <div
              className="text">
              In-Active
            </div>
            <div
              className="menu transition">
              <div
                aria-checked={false}
                aria-disabled={undefined}
                aria-selected={false}
                className="item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }>
                <i
                  aria-hidden="true"
                  className="green power icon" />
                Active
              </div>
              <div
                aria-checked={true}
                aria-disabled={undefined}
                aria-selected={true}
                className="active selected item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }>
                <i
                  aria-hidden="true"
                  className="red power icon" />
                In-Active
              </div>
              <div
                aria-checked={false}
                aria-disabled={undefined}
                aria-selected={false}
                className="item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }>
                <i
                  aria-hidden="true"
                  className="grey power icon" />
                None
              </div>
            </div>
          </div>
        </div>
        <div
          aria-busy={undefined}
          aria-disabled={undefined}
          aria-expanded={false}
          aria-multiselectable={undefined}
          className="ui pointing link item top right dropdown"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          style={
            Object {
              "float": "right",
              "fontSize": 20,
              "marginTop": -30,
              "paddingRight": 15,
            }
          }
          tabIndex={0}>
          <div
            className="text" />
          <i
            aria-hidden="true"
            className="angle down icon" />
          <div
            className="menu transition">
            <div
              aria-checked={undefined}
              aria-disabled={undefined}
              aria-selected={undefined}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "color": "#4183c4",
                }
              }>
              Edit
            </div>
            <div
              aria-checked={undefined}
              aria-disabled={undefined}
              aria-selected={undefined}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "color": "#4183c4",
                }
              }>
              Link
            </div>
            <div
              aria-checked={undefined}
              aria-disabled={undefined}
              aria-selected={undefined}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "color": "#4183c4",
                }
              }>
              Share
            </div>
            <div
              aria-checked={undefined}
              aria-disabled={undefined}
              aria-selected={undefined}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "color": "#4183c4",
                }
              }>
              Add to Logbooks
            </div>
            <div
              className="divider" />
            <div
              aria-checked={undefined}
              aria-disabled={undefined}
              aria-selected={undefined}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "color": "red",
                }
              }>
              Delete
            </div>
          </div>
        </div>
      </div>
      <div
        className="meta"
        style={
          Object {
            "display": "block",
            "fontSize": 14,
          }
        }>
        <span>
          Actions Taken
        </span>
      </div>
      <div
        className="text extra">
        Test comment 3
      </div>
      <div
        className="meta"
        style={
          Object {
            "marginTop": 5,
          }
        }>
        <span>
          3 Attachments
          <i
            aria-hidden="true"
            className="file icon" />
          <i
            aria-hidden="true"
            className="file icon" />
          <i
            aria-hidden="true"
            className="file icon" />
        </span>
      </div>
    </div>
  </div>
</div>
`;
